charts/platform-system/templates/flux/_repositories/helm-repository-istio.yaml

```diff
@@ -1,10 +1,16 @@
-{{ if and .Values.apps.istio.enabled (eq .Values.solution.name "flux") }}
+{{- if and .Values.apps.istio.enabled (eq .Values.solution.name "flux") -}}
+{{- $istioValues := .Values.apps.istio -}}
+{{- if $istioValues.source.create -}}
+---
 apiVersion: source.toolkit.fluxcd.io/v1beta1
-kind: HelmRepository
+kind: {{ default "HelmRepository" $istioValues.source.type }}
```

Comment:
This is intended for the `HelmRelease` resources. This will always be a `HelmRepository`. If someone wants to use a diff repo they should create the repo and provide the source ref to the release.